src/Microsoft.AspNetCore.OData/Formatter/ODataOutputFormatter.cs

```diff
@@ -194,7 +194,8 @@ public override void WriteResponseHeaders(OutputFormatterWriteContext context)
             // Set the character set.
             MediaTypeHeaderValue currentContentType = GetContentType(response.Headers[HeaderNames.ContentType].FirstOrDefault());
             RequestHeaders requestHeader = request.GetTypedHeaders();
-            if (requestHeader != null && requestHeader.AcceptCharset != null)
+            // Starting from ASP .NET Core 3.0 AcceptCharset returns an empty collection instead of null.
+            if (requestHeader?.AcceptCharset.Count > 0)
```

Comment:
```suggestion
            if (requestHeader?.AcceptCharset?.Count > 0)
```

Let's be compatible with both versions

Reply:
AFAIR this repo is for ASP .NET Core 3+. So it can't be ran with ASP .NET Core 2.x. This change make sense for OData/WebApi repo. Correct me if I'm wrong.

Reply:
I'd also lean in favor of the `AcceptCharset?.Count` check

Reply:
But the standard lib explicitly marks the properties that can be null as nullable in newer .NET versions e.g.: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.headers.requestheaders.cachecontrol?view=aspnetcore-6.0

I don't think that was the case for .NET Core 3+ so it's hard to tell from just looking at the property whether it's safe to assume it can't be null.